modules/ROOT/pages/batch-filters-and-batch-aggregator.adoc

```diff
@@ -190,34 +190,38 @@ Using random access, you can change a record's payload at any given index positi
+For a Batch Job, all records are first stored in a persistent queue before the Process phase begins, which means that the traditional threading model does not apply.
 
-Consider having 1 million records to place in a queue for a 3 step batch job. At least three million I/O operations occur as the runtime takes and requests each record as they move through the job's phases. +
-Performance requires having enough available memory to process the 16 threads in parallel, which means moving 1600 records from persistent storage into RAM. The larger your records and their quantity, the more available memory you need for batch processing.
+To improve performance, the runtime queues and schedules batch records in blocks of 100 records. This setting lessens the amount of I/O requests and improves an operation's load.
 
-Although, the standard model of 16 threads, with 100 records per batch job works for most use cases, consider three use cases where you might need to increase or decrease the block size:
+Batch jobs use Mule's thread pool.
+Each thread iterates through a block of up to 100 record. Then each block is queued back, and the process continues.
 
-* Assume you have 200 records to process through a batch job. With the default 100-record block size, Mule can only process two records in parallel at a time. If you request fewer than 101 records, then your processing becomes sequential. If you need to process heavy payloads, then queueing a hundred records demands a large amount of working memory.
-* Consider a batch job that needs to process images, and an average image size of 3 MB. You then have 100 blocks with payloads of 3 MB, being processed in 16 threads. Hence your default threading-profile setting would require around 4.6 GB of working memory just to keep the blocks in the queue. You should set a lower block size to distribute each payload through more jobs and lessen the load on your available memory.
-* Suppose having 5 million records with payloads so small that you can fit blocks of 500 records in your memory without problems. Setting a larger block size improves your batch job time without sacrificing working memory load.
+Consider having 1 million records to place in a queue for a 3-step Batch Job. At least three million I/O operations occur as the runtime takes and requests each record as they move through the job's phases.
 
-To take full advantage of this feature, you need to understand how the block sizes affect your batch job. Running comparative tests with different values and testing performance helps you find an optimum block size before moving this change into production.
+Performance requires having enough available memory to process all threads in parallel, which means moving records from persistent storage into RAM. The larger your records and their quantity, the more available memory you need for batch processing.
```

Comment:
this is absolutelly wrong. Discard all the copy edits and go back to the original mule 3/blog post content.